addons/account/views/account_invoice_view.xml

```diff
@@ -40,6 +40,14 @@
             </field>
         </record>
 
+        <record id="account_invoice_view_activity" model="ir.ui.view">
+            <field name="name">account.invoice.activity</field>
+            <field name="model">account.invoice</field>
+            <field name="arch" type="xml">
+                <activity string="Invoices" options='{"data_subtitle": "amount_total"}'/>
```

Comment:
After discussion with @ged-odoo:

The current have multiple drawback:
-images will be loaded by get activity data each time
-we need to customise get_activity_data when we need new field

In comparaison with kanban view:
-building url is better since image can be cached and loaded asynchronously.
-it is possible to customise template as we want for each model.
-no custom code in python.

More than that, taking inspiration from kanban-view will allow user to only learn one way to customize those tiles.

So, we would like to improve this a little:
Activity data arch should have field and template tags. 
A default template can be defined for most of the cases (just the name?)
A custom template can be defined (as in kanban views) to use the fields.
Image will be displayed with something similar to kanban_image (href forging) using _getImageURL (now in kanban_record.js, should be moved outside as a tool)
get_activity_data should be simplified: only a list of field will be given, and a read will be performed (instead of the custom logic depending on what options contains, managing placeholder and stuffs) 

Tell me if it is not clear of if you have any problem

Reply:
Sorry for taking so much time to react.  I just want to say that @Xavier-Do is absolutely right. The activity view should be customizable without requiring to add python code, and this can be done by having a template and a list of fields, just like in kanban view.